docs/core/encoding.md

```diff
@@ -274,6 +274,12 @@ Module should register interfaces using `InterfaceRegistry` which provides a mec
 
 In addition, an `UnpackInterfaces` phase should be introduced to deserialization to unpack interfaces before they're needed. Protobuf types that contain a protobuf `Any` either directly or via one of their members should implement the `UnpackInterfacesMessage` interface:
 
+### Can I customize a proto Message's String() method
+
+We should never customize a proto Message's `String()` method, i.e. we should never use `option (gogoproto.goproto_stringer) = false;`, doing otherwise leads to unexpected behaviour like returning wrong output or missing fields in the output.
```

Comment:
```suggestion
Using `option (gogoproto.goproto_stringer) = false;` in proto message definition leads to unexpected behaviour, like returning wrong output or having missing fields in the output.
For that reason it is strongly discouraged to use this option and attempt to customize a proto Message's `String()` method.
```

I think it fits more with the documentation style (we don't really use 'we shoud' anywhere)